### PR TITLE
Add failing test fixture for AddDefaultValueForUndefinedVariableRector (relates to #5253)

### PR DESCRIPTION
**Pull request by vladyslavstartsev**
relates to https://github.com/rectorphp/rector/issues/5253#issuecomment-763537452

# Failing Test for AddDefaultValueForUndefinedVariableRector

Based on https://getrector.org/demo/62ab1e88-6035-4fa6-87bb-c600dac418b2

### CODE REVIEW

File: rules/php56/tests/Rector/FunctionLike/AddDefaultValueForUndefinedVariableRector/Fixture/demo_file.php.inc [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+-----
+<?php
+
+namespace Rector\Php56\Tests\Rector\FunctionLike\AddDefaultValueForUndefinedVariableRector\Fixture;
+
+final class DemoFile
+{
+    public function run()
+    {
+        $a = [];
```
Comment (samsonasik) [inline rules/php56/tests/Rector/FunctionLike/AddDefaultValueForUndefinedVariableRector/Fixture/demo_file.php.inc:24]: I think `$a = null` (existing behaviour) is fine, ref https://phpstan.org/r/a62c48b7-46ee-4d63-9847-46f7c434e90b

Comment (vladyslavstartsev) [reply rules/php56/tests/Rector/FunctionLike/AddDefaultValueForUndefinedVariableRector/Fixture/demo_file.php.inc:24]: `null` is fine, but I think that `[]` is more familiar

Comment (TomasVotruba) [reply rules/php56/tests/Rector/FunctionLike/AddDefaultValueForUndefinedVariableRector/Fixture/demo_file.php.inc:24]: The default value should be `[]`. It seems like a bug in Rector

**Comment by samsonasik** [comment]
@vladyslavstartsev I cherry-picked your commits at PR #5293

**Pull request closed without merging**